server/src/services/commits.ts

```diff
@@ -60,4 +63,39 @@ const getAllCommits = async (
   return commitStorage.getAllCommits(filters);
 };
 
-export default {getAllCommits};
+/**
+ * Creates a new commit for the customer and the listing.
+ * An error would be thrown if the customer has reached their max num of ongoing commits,
+ * or if commit data provided is not valid.
+ * @param commitData Data of the commit to be added
+ */
+const addCommit = async (
+  commitData: CommitRequest
+): Promise<CommitResponse> => {
+  // Make sure that listing and customer exists
```

Comment:
Can there be a possibility that listing or customer does not exist and yet commit exists?

Is there a verification somewhere that ensure data consistency between COMMIT, LISTING, CUSTOMER and so on?

Reply:
@xlai20 
> Can there be a possibility that listing or customer does not exist and yet commit exists?

Line 76 and 77 are the checks done to prevent the commit from being created if customer or the listing does not exits. 
The datastore layer uses transaction so updating of commit count in listing and the adding of a new commit would be atomic.

Reply:
Got it. Does your Line 76 and 77 throw exception when there is no listing or customer of the requested ID? Can you point me to the location where it will throw an exception?

Reply:
@xlai20 Yep it does! The `datastore` layer throws an error when getting of a specified id fails. Can refer to the changes here: https://github.com/googleinterns/gpay-group-buy/pull/113/files#diff-f2fe6ca6032177087853d0c3ce45eedfR45-R51